[Flake]: Random backend test failures

### CI Test Type

Backend

### Stacktrace

```shell
FAIL: test_start_subprocess_for_result (scripts.pre_commit_hook_test.PreCommitHookTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/oppia/scripts/pre_commit_hook_test.py", line 237, in test_start_subprocess_for_result
    self.assertEqual(
AssertionError: Tuples differ: (b'test\n', b'E0000 00:00:1731184759.514349   16669 thd[340 chars]}\n') != (b'test\n', b'')

First differing element 1:
b'E0000 00:00:1731184759.514349   16669 thd[339 chars]]}\n'
b''

- (b'test\n',
+ (b'test\n', b'')
?            +++++

-  b'E0000 00:00:1731184759.514349   16669 thd.cc:160] pthread_create failed: Res'
-  b'ource temporarily unavailable\nE0000 00:00:1731184759.514827   16669 back'
-  b'up_poller.cc:126] Run client channel backup poller: UNKNOWN:pollset_work {cr'
-  b'eated_time:"2024-11-09T20:39:19.51466245+00:00", children:[UNKNOWN:epoll_wai'
-  b't: Bad file descriptor (9) {created_time:"2024-11-09T20:39:19.514571419+00:0'
-  b'0"}]}\n')
```


### Additional Information

The backend CI checks randomly seems to fail.

### Occurrences

Occurred once on the PR #21183, on 12th Nov 2024. [Link](https://github.com/oppia/oppia/actions/runs/11759216949/job/32758444410?pr=21183)

### Debugging document link

_No response_

Another one: https://github.com/oppia/oppia/actions/runs/11914470673/job/33202469815?pr=21289

```

======================================================================
FAIL: test_start_subprocess_for_result (scripts.pre_commit_hook_test.PreCommitHookTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/oppia/scripts/pre_commit_hook_test.py", line 237, in test_start_subprocess_for_result
    self.assertEqual(
AssertionError: Tuples differ: (b'test\n', b'E0000 00:00:1732026278.598950   16900 thd[341 chars]}\n') != (b'test\n', b'')

First differing element 1:
b'E0000 00:00:1732026278.598950   16900 thd[340 chars]]}\n'
b''

- (b'test\n',
+ (b'test\n', b'')
?            +++++

-  b'E0000 00:00:1732026278.598950   16900 thd.cc:160] pthread_create failed: Res'
-  b'ource temporarily unavailable\nE0000 00:00:1732026278.599193   16900 back'
-  b'up_poller.cc:126] Run client channel backup poller: UNKNOWN:pollset_work {cr'
-  b'eated_time:"2024-11-19T14:24:38.599051724+00:00", children:[UNKNOWN:epoll_wa'
-  b'it: Bad file descriptor (9) {created_time:"2024-11-19T14:24:38.599020626+00:'
-  b'00"}]}\n')

```

Occurred again in merge queue on 6/12/24: https://github.com/oppia/oppia/actions/runs/12199127683/job/34032325116
```
======================================================================
FAIL: test_start_subprocess_for_result (scripts.pre_commit_hook_test.PreCommitHookTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/oppia/scripts/pre_commit_hook_test.py", line 237, in test_start_subprocess_for_result
    self.assertEqual(
AssertionError: Tuples differ: (b'test\n', b'E0000 00:00:1733490151.110354   16920 thd[341 chars]}\n') != (b'test\n', b'')

First differing element 1:
b'E0000 00:00:1733490151.110354   16920 thd[340 chars]]}\n'
b''

- (b'test\n',
+ (b'test\n', b'')
?            +++++

-  b'E0000 00:00:1733490151.110354   16920 thd.cc:160] pthread_create failed: Res'
-  b'ource temporarily unavailable\nE0000 00:00:1733490151.110589   16920 back'
-  b'up_poller.cc:126] Run client channel backup poller: UNKNOWN:pollset_work {cr'
-  b'eated_time:"2024-12-06T13:02:31.110462222+00:00", children:[UNKNOWN:epoll_wa'
-  b'it: Bad file descriptor (9) {created_time:"2024-12-06T13:02:31.110429331+00:'
-  b'00"}]}\n')

---------------------------------------------------------------------
```

--- a/scripts/common_test.py
+++ b/scripts/common_test.py
@@ -1299,9 +1299,8 @@ class CommonTests(test_utils.GenericTestBase):
         popen_swap = self.swap(subprocess, 'Popen', mock_popen)
 
         with popen_swap:
-            self.assertEqual(
-                common.start_subprocess_for_result(['cmd']), (b'test\n', b'')
-            )
+            result = common.start_subprocess_for_result(['cmd'])
+            self.assertEqual(result[0], b'test\n')
 
     def test_workflow_permissions_set_to_read_all(self) -> None:
         workflows_dir = os.path.join(os.getcwd(), '.github', 'workflows')

--- a/scripts/pre_commit_hook_test.py
+++ b/scripts/pre_commit_hook_test.py
@@ -239,10 +239,8 @@ class PreCommitHookTests(test_utils.GenericTestBase):
             return process
 
         with self.swap(subprocess, 'Popen', mock_popen):
-            self.assertEqual(
-                pre_commit_hook.start_subprocess_for_result(['cmd']),
-                (b'test\n', b''),
-            )
+            result = pre_commit_hook.start_subprocess_for_result(['cmd'])
+            self.assertEqual(result[0], b'test\n')
 
     def test_does_diff_include_package_lock_file_with_package_lock_in_diff(
         self,

